File: resources/exercises/turmites/langton/LangtonEntity.py


# BEGIN TEMPLATE
def step():
	# BEGIN SOLUTION
	if getGroundColor() == Color.white:
		right()
		setBrushColor(Color.black)
		brushDown()
		brushUp()
		forward()
	else:
		left()
		setBrushColor(Color.white)
		brushDown()
		brushUp()
		forward()
	# END SOLUTION
# END TEMPLATE
		
for i in range(getParam(0)):
	step()
	stepDone()

File: resources/exercises/turmites/langton/LangtonEntity.c
//RemoteBuggle

void step();


void run(){
	int nbSteps = getParam();
	int i;
	for (i=0;i<nbSteps;i++) {
		step();
		stepDone();
	}
}

/* BEGIN TEMPLATE */
void step() {
	/* BEGIN SOLUTION */
	if (getGroundColor()==white) {
		right();

		setBrushColor(black);
		brushDown();
		brushUp();

		forward(1);
	} else {
		left();

		setBrushColor(white);
		brushDown();
		brushUp();

		forward(1);
	}
	/* END SOLUTION */
}
/* END TEMPLATE */
